client/src/java/com/zimbra/client/ZPop3DataSource.java

```diff
@@ -68,7 +66,7 @@ public Element toElement(Element parent) {
         src.addAttribute(MailConstants.A_DS_PASSWORD, data.getPassword());
         src.addAttribute(MailConstants.A_FOLDER, data.getFolderId());
         src.addAttribute(MailConstants.A_DS_CONNECTION_TYPE, data.getConnectionType().name());
-        src.addAttribute(MailConstants.A_DS_LEAVE_ON_SERVER, data.isLeaveOnServer());
+        src.addAttribute(MailConstants.A_DS_LEAVE_ON_SERVER, getData().isLeaveOnServer());
```

Comment:
The contents of the `leaveOnServer` method below imply that the last arg in this can be null which will lead to an NPE.  Suggest just using the `leaveOnServer` method which handles that.